docs/src/reference/gremlin-variants.asciidoc

```diff
@@ -1897,3 +1897,300 @@ pip install gremlinpython
+
+[source,go]
+----
+resultSet, err := client.Submit("g.V().repeat(both()).times(100)", map[string]interface{}{"evaluationTimeout": 5000})
+----
+
+The following options are allowed on a per-request basis in this fashion: `batchSize`, `requestId`, `userAgent` and
+`evaluationTimeout` (formerly `scriptEvaluationTimeout` which is also supported but now deprecated).
+
+IMPORTANT: The preferred method for setting a per-request timeout for scripts is demonstrated above, but those familiar
+with bytecode may try `g.with("evaluationTimeout", 500)` within a script. Scripts with multiple traversals and multiple
+timeouts will be interpreted as a sum of all timeouts identified in the script for that request.
+
+[source,go]
+----
+resultSet, err := client.Submit("g.with('evaluationTimeout', 500).addV().iterate();"+
+  "g.addV().iterate();"+
+  "g.with('evaluationTimeout', 500).addV();", map[string]interface{}{"evaluationTimeout": 500})
+results, err := resultSet.All()
+----
+
+In the above example, defines a timeout of 500 milliseconds, but the script has three traversals with
+two internal settings for the timeout using `with()`. The request timeout used by the server will therefore be 1000
+milliseconds (overriding the 500 which itself was an override for whatever configuration was on the server).
+
+[[gremlin-go-differences]]
+=== Differences
+
+In situations where Go reserved words and global functions overlap with standard Gremlin steps and tokens, those
+bits of conflicting Gremlin get an underscore appended as a suffix. In addition, all function names start with a 
+capital letter in order to be public:
+
+*Steps* - <<and-step,And()>>, <<as-step,As()>>, <<filter-step,Filter()>>, <<from-step,From()>>, <<id-step,Id()>>,
+<<is-step,Is()>>, <<in-step,In()>>, <<max-step,Max()>>, <<min-step,Min()>>, <<not-step,Not()>>, <<or-step,Or()>>,
+<<range-step,Range()>>, <<sum-step,Sum()>>, <<with-step,With()>>
+
+*Tokens* - <<a-note-on-scopes,Scope.Global>>, <<a-note-on-scopes,Scope.Local>>
+
+*Statics* - <<__, T__>> // todo: impove 
+
+[[gremlin-go-limitations]]
+=== Limitations
+
+* There is no default `set` type in Go. Any set type code from server will be deserialized into slices with the list 
+type implementation. To input a set into Gremlin-Go, a custom struct which implements the `gremlingo.Set` interface 
+will be serialized as a set. `gremlingo.NewSimpleSet` is a basic implementation of a set that is provided by Gremlin-Go 
+that can be used to fulfill the `gremlingo.Set` interface if desired.
+* The `path` data type serialization is currently incomplete as labels are represented as list of lists instead of list 
+of sets. Fully functional Path serialization will be implemented when set is implemented in the next milestone. Path 
+can be successfully deserialized.
+* Traversal step functions currently take `string` arguments with double quotes only. Operations using Gremlin keywords,
```

Comment:
is this still true? how are the gherkin tests passing - we surely have a `by(T.label)` somewhere in those tests?

Reply:
There are outdated, will remove